lahja/endpoint.py

```diff
@@ -266,35 +238,49 @@ class Endpoint(BaseEndpoint):
     The :class:`~lahja.endpoint.Endpoint` enables communication between different processes
     as well as within a single process via various event-driven APIs.
     """
-
-    _name: str
-    _ipc_path: pathlib.Path
+    __slots__ = (
+        'ipc_path',
+        '_child_tasks',
+        '_full_connections',
+        '_futures',
+        '_half_connections',
+        '_handlers',
+        '_internal_queue',
+        '_queues',
+        '_receiving_loop_running',
+        '_receiving_queue',
+        '_receiving_queue',
+        '_running',
+        '_server_running',
+        '_serving',
+        '_who_are_you_stream_handler_running',
+    )
+
+    ipc_path: pathlib.Path
 
     _receiving_queue: 'asyncio.Queue[Tuple[Union[bytes, BaseEvent], Optional[BroadcastConfig]]]'
     _receiving_loop_running: asyncio.Event
 
-    _internal_queue: 'asyncio.Queue[Tuple[BaseEvent, Optional[BroadcastConfig]]]'
-    _internal_loop_running: asyncio.Event
-
     _server_running: asyncio.Event
 
-    _loop: Optional[asyncio.AbstractEventLoop]
+    _loop: Optional[asyncio.AbstractEventLoop] = None
+
+    def __init__(self, name: str) -> None:
+        self.name = name
+
+        # connections that we expect to both send and receive events.
+        self._full_connections: Dict[str, RemoteEndpoint] = {}
+        # connections that we will only receive events from.
+        self._half_connections: Set[RemoteEndpoint] = set()
 
-    def __init__(self) -> None:
-        self._outbound_connections: Dict[str, OutboundConnection] = {}
-        self._inbound_connections: Set[InboundConnection] = set()
         self._futures: Dict[Optional[str], 'asyncio.Future[BaseEvent]'] = {}
         self._handler: Dict[Type[BaseEvent], List[Callable[[BaseEvent], Any]]] = {}
         self._queues: Dict[Type[BaseEvent], List['asyncio.Queue[BaseEvent]']] = {}
 
         self._child_tasks: Set['asyncio.Future[Any]'] = set()
 
         self._running = False
-        self._loop = None
-
-    @property
-    def ipc_path(self) -> pathlib.Path:
-        return self._ipc_path
+        self._serving = False
```

Comment:
This is a simple stand-alone, ensuring that running and serving are independent concepts.